private/protocol/json/jsonutil/unmarshal.go

```diff
@@ -211,3 +227,33 @@ func unmarshalScalar(value reflect.Value, data interface{}, tag reflect.StructTa
 	}
 	return nil
 }
+
+func timeFromUnixString(d string) (time.Time, error) {
+	var secs, nsecs int64
+	if dot := strings.Index(d, "."); dot != -1 {
+		var err error
+		secs, err = strconv.ParseInt(d[:dot], 10, 64)
```

Comment:
So, I tried that originally — I hate writing so much string parsing!

But here's the question: what should nsecs be for `1501917228.309`?  I'd assume it should be 309000000.  However, if you just parse as a float, Modf, multiply by 1e9, and convert to integer, you get 309000015 due to floating-point precision issues: https://play.golang.org/p/ytEH62Alfp

I alluded to this in the PR description (though I incorrectly said "rounding" rather than precision) but should have been more explicit.

I'm not sure what the best step from here is:
- Ignore precision issues
- Do normal float parsing but also count the number of post-decimal digits and round based on that?
- Apply the logic here if `^\d+(\.\d+)?$` and do something different if exponent is applied?
- Extend the logic here to handle exponents (though it's not super clear to me what that should mean)?

Thoughts?

Also I guess I should handle a `+` prefix (and a `-` too, for pre-1970 times?)